pkgs/tools/misc/moon-phases/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "moon-phases";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "mirrorwitch";
+    repo = "moon-phases";
+    rev = "v0.3.3";
+    hash = "sha256-WeIElPuTLKaR2A64W1I7ITMsdEkin2bOYQl9KDtNmBI=";
+  };
```

Comment:
Since this crate is released on https://crates.io/crates/moon-phases, you should retrieve it from there using `fetchCrate`:

```nix
  src = fetchCrate {
    inherit pname version;
    hash = "…";
  };
```